src/foam/lib/json/Outputter.java

```diff
@@ -96,19 +97,38 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
-    writer_.append("\"");
-    writer_.append(escape(s));
-    writer_.append("\"");
+    if ( multiLineOutput_ && s.indexOf('\n') >= 0 ) {
+      writer_.append("\n  ");
+      writer_.append("\"");
+      writer_.append("\"\"\n    ");
+      s = ltrim(s, 4);
+      writer_.append(escapeMultiline(s));
+      writer_.append("\n  \"\"");
+      writer_.append("\"");
+    }
+    else {
+      writer_.append("\"");
+      writer_.append(escape(s));
+      writer_.append("\"");
+    }
   }
 
   public String escape(String s) {
-    return s
-      .replace("\\", "\\\\")
+    return s.replace("\\", "\\\\")
```

Comment:
The previous line break of
```
return s
    .replace()
    .replace()
...
```

is easier to read IMO, as all the replace() calls are lined up.